src/main/java/org/openrewrite/java/spring/SeparateApplicationPropertiesByProfile.java

```diff
@@ -0,0 +1,179 @@
+        if (acc.propertyFileContent.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Set<SourceFile> newApplicationPropertiesFiles = new HashSet<>();
+
+        for (Map.Entry<String, List<Properties.Content>> entry : acc.propertyFileContent.entrySet()) {
+            if (!acc.fileNameToFilePath.containsKey(entry.getKey())) {
+                newApplicationPropertiesFiles.
+                        add(new CreatePropertiesFile(acc.pathToApplicationProperties + entry.getKey(), "", null).
+                                generate(new AtomicBoolean(true), ctx).
+                                iterator().
+                                next());
+            }
+        }
+
+        return newApplicationPropertiesFiles;
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor(Accumulator acc) {
+        return new PropertiesVisitor<ExecutionContext>() {
+            @Override
+            public Properties visitFile(Properties.File file, ExecutionContext ctx) {
+                if (acc.propertyFileContent.isEmpty()) {
+                    return file;
+                }
+
+                String[] filePathArray = file.getSourcePath().toString().split("/");
+                String fileName = filePathArray[filePathArray.length - 1];
+
+                return fileName.matches("application.properties") ? deleteFromApplicationProperties(file) :
+                        appendToExistingPropertiesFile(file, acc.propertyFileContent.get(fileName));
+            }
+        };
+    }
+
+    private Properties appendToExistingPropertiesFile(Properties.File file, List<Properties.Content> contentToAppend) {
+        return file.withContent(
+                Stream.concat(file.getContent().stream(), contentToAppend.stream()).
+                        collect(Collectors.toList()));
+    }
+
+    private Properties deleteFromApplicationProperties(Properties.File applicationProperties) {
+        List<Properties.Content> newContent = new ArrayList<>();
+        for (Properties.Content c : applicationProperties.getContent()) {
+            if (isSeparator(c)) {
+                break;
+            }
+            newContent.add(c);
+        }
+        return applicationProperties.getContent().equals(newContent) ? applicationProperties :
+                applicationProperties.withContent(newContent);
+    }
+
+    private Map<String, List<Properties.Content>> getNewApplicationPropertyFileInfo(List<Properties.Content> contentList) {
+        Map<String, List<Properties.Content>> map = new HashMap<>();
+        int index = 0;
+        while (index < contentList.size()) {
+            if (isSeparator(contentList.get(index))) {
+                List<Properties.Content> newContent = getContentForNewFile(contentList, ++index);
+                map.put("application-" + ((Properties.Entry) newContent.get(0)).getValue().getText() + ".properties",
+                        newContent.subList(1, newContent.size()));
```

Comment:
I think we should use the full target file name as key here, since there could be a multi module project that could clash.
```
parent
  - childA
    - src/main/resource/application.properties
  - childB
    - src/main/resource/application.properties
```

Reply:
I believe the code already adjusts for files within nested folders like your example (line 206 of the testing file). I also do not think that a project can have multiple application.properties files. 

The method above - getNewApplicationPropertyFileInfo is simply for getting the content of each application-env.properties file from application.properties. Application-env.properties files that don't exist will, by default, be placed in the same folder as the application.properties file.

Reply:
hi! I've just pushed a test to highlight what I meant; https://github.com/openrewrite/rewrite-spring/pull/563/commits/4f232974640180041c78307cde8c156c1fef9538
The test at line 206 only had a single folder structure, not two separate.

Given the scale at which recipes can run I think we should account for such a scenario.